### DISCUSSION TITLE
Allow the option to sync the app settings between devices.

### DISCUSSION THREAD

**Discussion by streetkid1215** [ente-io/ente#496] [Enhancements]
I have noticed that the settings need to be set for each device I log into. I have to refer to existing devices to make sure the settings match also noticed if the app is reset or logged out for any reason, the settings are also forgotten. 

Is it possible to allow the applied settings to sync when being changed on an existing device attached to the account? Perhaps there could also be an initial option or wizard to import the existing settings when completing a new login into a device.

**Comment by vishnukvmd** [comment]
The `keys` and corresponding `values` to be exported are within
https://github.com/ente-io/auth/blob/8d82851741f0374deac1ebe0900af715702fc8fb/lib/services/preference_service.dart#L5

Persisting these settings upstream is an option, but they'd have to be stored e2ee as well.

An easier way forward would be as @streetkid1215 suggested, by adding a separate `preferences` entry within the exported file that is a map of these `keys` and `values`.

This will be unblocked once https://github.com/ente-io/auth/pull/382 lands, since it introduces exports in the format of a JSON file.

`PreferenceService` could then expose two APIs
1. `fetchPreferences` to return this map
2. `applyPreferences` to apply this map

**Comment by jellemdekker** [comment]
For reference, #382 has been superseded by #386.
